include/app_memory/app_memdomain.h

```diff
@@ -77,12 +86,30 @@ struct app_region {
 	{ \
 		name.dmem_start = (char *)&data_smem_##name; \
 		name.bmem_start = (char *)&data_smem_##name##b; \
+		name.smem_size = data_smem_##name##_size; \
 		sys_dlist_append(&app_mem_list, &name.lnode); \
 		mem_domain_##name.start = (u32_t) name.dmem_start; \
 		mem_domain_##name.attr = K_MEM_PARTITION_P_RW_U_RW; \
 		name.partition = &mem_domain_##name; \
 	}
-
+#else
+#define appmem_partition(name) \
+	extern char *data_smem_##name; \
+	extern char *data_smem_##name##b; \
+	_app_dmem_pad(name) char name##_dmem_pad; \
+	_app_bmem_pad(name) char name##_bmem_pad; \
+	__kernel struct k_mem_partition mem_domain_##name; \
+	__kernel struct app_region name; \
+	static inline void appmem_init_part_##name(void) \
+	{ \
+		name.dmem_start = (char *)&data_smem_##name; \
+		name.bmem_start = (char *)&data_smem_##name##b; \
+		sys_dlist_append(&app_mem_list, &name.lnode); \
+		mem_domain_##name.start = (u32_t) name.dmem_start; \
+		mem_domain_##name.attr = K_MEM_PARTITION_P_RW_U_RW; \
+		name.partition = &mem_domain_##name; \
+	}
+#endif	/* CONFIG_MPU_REQUIRES_POWER_OF_TWO_ALIGNMENT */
```

Comment:
Could you please find a way to avoid maintaining two copies of appmem_partition.